lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -651,18 +651,18 @@ defmodule RealtimeWeb.RealtimeChannel do
         role: claims["role"]
       })
 
-    {:ok, socket} = Authorization.get_authorizations(socket, db_conn, authorization_context)
-
-    cond do
-      match?(%Policies{broadcast: %BroadcastPolicies{read: false}}, socket.assigns.policies) ->
-        {:error, "You do not have permissions to read from this Topic"}
+    with {:ok, socket} <- Authorization.get_authorizations(socket, db_conn, authorization_context) do
+      cond do
+        match?(%Policies{broadcast: %BroadcastPolicies{read: false}}, socket.assigns.policies) ->
+          {:error, "You do not have permissions to read from this Topic"}
 
-      using_broadcast? &&
-          match?(%Policies{broadcast: %BroadcastPolicies{read: false}}, socket.assigns.policies) ->
-        {:error, "You do not have permissions to read Broadcast messages from this channel"}
+        using_broadcast? &&
+            match?(%Policies{broadcast: %BroadcastPolicies{read: false}}, socket.assigns.policies) ->
+          {:error, "You do not have permissions to read Broadcast messages from this channel"}
 
-      true ->
-        {:ok, socket}
+        true ->
+          {:ok, socket}
+      end
```

Comment:
we should have an `else` block to catch `Authorization.get_authorizations/3` errors.

Reply:
covered in https://github.com/supabase/realtime/pull/1120